### PR TITLE
DEV-957 update website events and sdn tutorials

### PR DESCRIPTION
**Pull request by teye**

### CODE REVIEW

File: src/main/java/sg/ncl/MainController.java [hunk 1]
```diff
@@ -166,6 +166,12 @@ public String about() {
         return "about";
     }
 
+    @RequestMapping("/BNVirtusage")
+    public String BNVirtusage() {
```
Comment (ZHANGChunwang) [inline src/main/java/sg/ncl/MainController.java]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)

**Comment by ZHANGChunwang** [comment]
SonarQube analysis reported 6 issues
* ![CRITICAL][CRITICAL] 4 critical
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [MainController.java#L864](https://github.com/nus-ncl/service-web/blob/e9796b040baf89a6d55514f64de2134d76bcbd9b/src/main/java/sg/ncl/MainController.java#L864): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [MainController.java#L1804](https://github.com/nus-ncl/service-web/blob/e9796b040baf89a6d55514f64de2134d76bcbd9b/src/main/java/sg/ncl/MainController.java#L1804): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [MainController.java#L1893](https://github.com/nus-ncl/service-web/blob/e9796b040baf89a6d55514f64de2134d76bcbd9b/src/main/java/sg/ncl/MainController.java#L1893): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [MainController.java#L3026](https://github.com/nus-ncl/service-web/blob/e9796b040baf89a6d55514f64de2134d76bcbd9b/src/main/java/sg/ncl/MainController.java#L3026): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![MINOR][MINOR] [MainController.java#L1358](https://github.com/nus-ncl/service-web/blob/e9796b040baf89a6d55514f64de2134d76bcbd9b/src/main/java/sg/ncl/MainController.java#L1358): Replace this "filter().findFirst().isPresent()" chain with "anyMatch()" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4034)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by ZHANGChunwang** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [MainController.java#L858](https://github.com/nus-ncl/service-web/blob/8d0a533256551fb7ad951a3b02fceca31c29666d/src/main/java/sg/ncl/MainController.java#L858): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [MainController.java#L1798](https://github.com/nus-ncl/service-web/blob/8d0a533256551fb7ad951a3b02fceca31c29666d/src/main/java/sg/ncl/MainController.java#L1798): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [MainController.java#L1887](https://github.com/nus-ncl/service-web/blob/8d0a533256551fb7ad951a3b02fceca31c29666d/src/main/java/sg/ncl/MainController.java#L1887): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [MainController.java#L3020](https://github.com/nus-ncl/service-web/blob/8d0a533256551fb7ad951a3b02fceca31c29666d/src/main/java/sg/ncl/MainController.java#L3020): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![MINOR][MINOR] [MainController.java#L1352](https://github.com/nus-ncl/service-web/blob/8d0a533256551fb7ad951a3b02fceca31c29666d/src/main/java/sg/ncl/MainController.java#L1352): Replace this "filter().findFirst().isPresent()" chain with "anyMatch()" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4034)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/sg/ncl/MainController.java [hunk 2]
```diff
@@ -855,12 +855,12 @@ private void registerUserToDeter(JSONObject mainObject) throws
 
                 ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
 
-                final String errorPrefix = "Error: ";
+                final String ERROR_PREFIX = "Error: ";
```
Comment (ZHANGChunwang) [inline src/main/java/sg/ncl/MainController.java]: MAJOR Rename "ERROR_PREFIX" which hides the field declared at line 80. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)

**Comment by ZHANGChunwang** [comment]
SonarQube analysis reported 5 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [MainController.java#L1798](https://github.com/nus-ncl/service-web/blob/a24012b349f890954f7b72a0226a692e93ce2023/src/main/java/sg/ncl/MainController.java#L1798): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [MainController.java#L1887](https://github.com/nus-ncl/service-web/blob/a24012b349f890954f7b72a0226a692e93ce2023/src/main/java/sg/ncl/MainController.java#L1887): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [MainController.java#L3020](https://github.com/nus-ncl/service-web/blob/a24012b349f890954f7b72a0226a692e93ce2023/src/main/java/sg/ncl/MainController.java#L3020): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![MINOR][MINOR] [MainController.java#L1352](https://github.com/nus-ncl/service-web/blob/a24012b349f890954f7b72a0226a692e93ce2023/src/main/java/sg/ncl/MainController.java#L1352): Replace this "filter().findFirst().isPresent()" chain with "anyMatch()" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4034)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by ZHANGChunwang** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [MainController.java#L1885](https://github.com/nus-ncl/service-web/blob/138e3e091620db9d44622af1b98d3830927c6d18/src/main/java/sg/ncl/MainController.java#L1885): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [MainController.java#L3018](https://github.com/nus-ncl/service-web/blob/138e3e091620db9d44622af1b98d3830927c6d18/src/main/java/sg/ncl/MainController.java#L3018): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![MINOR][MINOR] [MainController.java#L1350](https://github.com/nus-ncl/service-web/blob/138e3e091620db9d44622af1b98d3830927c6d18/src/main/java/sg/ncl/MainController.java#L1350): Replace this "filter().findFirst().isPresent()" chain with "anyMatch()" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4034)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by ZHANGChunwang** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [MainController.java#L1350](https://github.com/nus-ncl/service-web/blob/378592afaa0ed6b3bcccd67989b87e554e043c7e/src/main/java/sg/ncl/MainController.java#L1350): Replace this "filter().findFirst().isPresent()" chain with "anyMatch()" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4034)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by nghianja** [comment]
Changes required:

1. event.html, line 45: Remove the first word "Showcase" in the paragraph.

1. index.html: Remove commented lines to prevent sending unnecessary text.

**Comment by ZHANGChunwang** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [MainController.java#L1350](https://github.com/nus-ncl/service-web/blob/d727d0f71e46e269963ac22684db0ea5523ee17b/src/main/java/sg/ncl/MainController.java#L1350): Replace this "filter().findFirst().isPresent()" chain with "anyMatch()" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4034)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**